File: algorithms/Strings/Easy/UniqueMorseCodeWords/VSDUniqueMorseCodeWords.py


def uniqueMorseRepresentations(words:list[str])->int:
    morsedict=dict()
    morse=[".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."]
    j=0
    for i in range(97,97+26):
        morsedict[chr(i)]=morse[j]
        j+=1
    
    lst=set()
    for word in words:
        answer=""
        for char in word:
            answer+=morsedict[char]
        lst.add(answer)
    return len(lst)    

words = ["gin","zen","gig","msg"]
print(uniqueMorseRepresentations(words))

File: algorithms/Strings/Easy/UniqueMorseCodeWords/VSDUniqueMorseCodeWords.java
import java.util.HashSet;

public class Main{

    static String[] morse_code={".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."};

    static int code(String[] words){
        HashSet<String> lst=new HashSet<>();
        String ans;
        for(String word:words){
            ans="";
            for(char ch:word.toCharArray()){
                ans+=morse_code[ch-97];

            }
            lst.add(ans);
        }
        return lst.size();
    }
    public static void main(String[] args) {
        String[] words={"gin","zen","gig","msg"};
        System.out.println(code(words));
    }
}